main.tf

```diff
@@ -16,8 +16,9 @@
 
 
 locals {
-  address = var.create_address ? join("", google_compute_global_address.default.*.address) : var.address
-  url_map = var.create_url_map ? join("", google_compute_url_map.default.*.self_link) : var.url_map
+  address                    = var.create_address ? join("", google_compute_global_address.default.*.address) : var.address
+  url_map                    = var.create_url_map ? join("", google_compute_url_map.default.*.self_link) : var.url_map
+  backends_with_healthchecks = [for backend in var.backends : backend if backend["health_check"] != null]
```

Comment:
Was this intended to be a list? We may need to use a set because you are using it with a resource `for_each`

Reply:
I think it transforms one map into another, but I am unfamiliar with HCL.

According to https://www.hashicorp.com/blog/hashicorp-terraform-0-12-preview-for-and-for-each/ it can indeed take a map and return another map.

`backends` is a map according to https://github.com/terraform-google-modules/terraform-google-lb-http/blob/ca8a56700f0b03f8bf4353fde182ff72673a2a5d/variables.tf#L69-L71

Reply:
@naseemkullah since you are doing `[......]` this will return a list. I took a look the backend object and I think 
`{for backend_name, backend in var.backends : backend_name => backend if backend["health_check"] != null}` makes sense syntactically. @morgante WDYT?

Reply:
Yes we should keep this as a map.